### ISSUE TITLE
Multiple denylist patterns are not applied to CDATA validation

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#4319]
## Bug Description

It turns out that CDATA can have multiple `blacklisted_cdata_regex` constraints, for [example](https://github.com/ampproject/amphtml/blob/911661bd0d2f07bcf9cca9312fea34175ace32af/validator/validator-main.protoascii#L1102-L1115):

```
    blacklisted_cdata_regex: {
      regex: "<!--"
      error_message: "html comments"
    }
    # These regex blacklists are temporary hacks to validate essential CSS
    # rules. They will be replaced later with more principled solutions.
    blacklisted_cdata_regex: {
      regex: "(^|\\W)i-amphtml-"
      error_message: "CSS i-amphtml- name prefix"
    }
    blacklisted_cdata_regex: {
      regex: "!\\s*important"
      error_message: "CSS !important"
    }
```

However, the Python spec parser is only capturing one:

https://github.com/ampproject/amp-wp/blob/b059e63c1de983a94d598142fded3c6758b0d0b9/includes/sanitizers/class-amp-allowed-tags-generated.php#L15398-L15403

And the sanitizer is only expecting one:

https://github.com/ampproject/amp-wp/blob/b059e63c1de983a94d598142fded3c6758b0d0b9/includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php#L902-L918

This means at present CSS selectors that contain `i-amphtml-*` will not get caught by the sanitizer, even though they are invalid. Related: https://github.com/ampproject/amp-wp/issues/771.

We need to fix how the spec is parsed for `blacklisted_cdata_regex`. We should also check to see if this same thing is happening for other properties. At first I thought `blacklisted_value_regex` would be a candidate, but it seems to already be changed to concatenate all denied patterns into a single regex.

## Expected Behaviour

Multiple `blacklisted_cdata_regex` constraints should be captured and applied during sanitization/validation.

## Additional context

 - WordPress version:
 - Plugin version:
 - Gutenberg plugin version (if applicable):
 - AMP plugin template mode:
 - PHP version:
 - OS:
 - Browser: [e.g. chrome, safari]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by westonruter** [comment]
This is being fixed as part of #4548.

**Comment by kienstra** [comment]
OK, thanks!
